File: Javascript/26.删除排序数组中的重复项.js
/*
 * @lc app=leetcode.cn id=26 lang=javascript
 *
 * [26] 删除排序数组中的重复项
 */
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    var temp = {}                 //创建一个新对象
    // var arr = [];                 //创建一个空数组
    var count=0;
    len = nums.length;
    for (var i = 0; i < len; i++) {
        if (!temp[nums[i]]) {                //第一次都为undefined,所以要为其赋值-->!undefined继续执行。
            temp[nums[i]] = "abc";          //为第一次变为属性名的赋值
            // arr.push(nums[i]);           //放到新数组上去
            count++;
        }
    }
    return count;              //返回去重后的数组
};

File: Clanguage/26.删除排序数组中的重复项.c
/*
 * @lc app=leetcode.cn id=26 lang=c
 *
 * [26] 删除排序数组中的重复项
 * 双指针
 */


int removeDuplicates(int* nums, int numsSize){
 if(numsSize==0){
     return 0;
 }
 int i=0; //b不同数值计数器。移动指针
 int j=0; //块指针
 for(j=0;j<numsSize;j++){
     if(nums[j]!=nums[i]){ //相等时，就跳过重复项，直接j++,不i++
         i++;             //不相等等时，计数并移位
         nums[i]=nums[j];  //将j位的值赋给i的下一位（i的下一位等于i位）--》删出了i++位和i位相同的元素
     }
 }
 return i+1;
}



